### PR TITLE
Add noindex meta tag 

### PR DESCRIPTION
**Pull request by camigthompson**
Add the `noindex` meta tag to the user profiles page (for locked users) and to the report abuse page and package statistics pages for all packages. 

Addresses https://github.com/NuGet/Engineering/issues/4457

### CODE REVIEW

File: src/NuGetGallery/Views/Statistics/PackageVersions.cshtml [hunk 1]
```diff
@@ -14,6 +14,8 @@
     @:data-bind="visible: !showAllPackageDownloads()"
 }
 
+<meta name="robots" content="noindex">
```
Comment (joelverhagen) [inline src/NuGetGallery/Views/Statistics/PackageVersions.cshtml:17]: I believe this stuff needs to be in the `<head>` section of the HTML. The way we do it on package details page is here: https://github.com/NuGet/NuGetGallery/blob/2edeac8ab27019d4bc28bf69c8019b5276442f11/src/NuGetGallery/Views/Packages/DisplayPackage.cshtml#L8

**Review by joelverhagen — APPROVED** [review]

**Review by shishirx34 — APPROVED** [review]

**Comment by camigthompson** [comment]
Performed some validation and confirmed that the `noindex` meta tag appears on the html of the changed pages. 

image

image

image

image

**Comment by shishirx34** [comment]
BTW, if you are not aware please `Squash and merge` the pull request when you are done so that the branch commits don't overflow into the dev commit history.

**Review by dannyjdev — APPROVED** [review]

**Comment by joelverhagen** [comment]
> Performed some validation and confirmed that the `noindex` meta tag appears on the html of the changed pages.

@camigthompson - great validations! Nice work!

**Pull request merged**